### PR TITLE
Add exported sub-order types for ease of use by the community

### PR DESCRIPTION
**Pull request by jdgriffith**
I was wondering if you would accept changes such as this. People might be using your package for complex needs and sometimes it's beneficial to pull in specific types into your project for nuanced logic before submitting an order. Another benefit here is it makes the `Order` and `SavedOrder` much easier to read and reason about.

Thanks again for all your hard work in the initial documentation of these types! It's very helpful. Let me know if there is anything else I need to do to make this merge ready.

### CODE REVIEW

File: src/order.ts [hunk 1]
```diff
@@ -1,643 +1,230 @@
...
+  | 'MARKET_ON_CLOSE'
+  | 'EXERCISE'
+  | 'TRAILING_STOP_LIMIT'
+  | 'NET_DEBIT'
+  | 'NET_CREDIT'
+  | 'NET_ZERO';
+
+export type CancelTime = { date?: string; shortFormat?: boolean };
+
+export type ComplexOrdertStrategyType =
```
Comment (MorpheusNephew) [inline src/order.ts:18]: Thanks for the changes @jdgriffith I really appreciate it. The only thing here is removing the t between Order and Strategy

**Review by MorpheusNephew — CHANGES_REQUESTED** [review]

**Review by MorpheusNephew — APPROVED** [review]

File: src/order.ts [hunk 2]
```diff
@@ -1,643 +1,230 @@
...
   closeTime?: string;
   tag?: string;
   accountId?: number;
   orderActivityCollection?: {
-    activityType?: 'EXECUTION' | 'ORDER_ACTION';
+    activityType?: ActivityType;
   }[];
-  replacingOrderCollection?: any[];
-  childOrderStrategies?: any[];
+  replacingOrderCollection?: Order[];
```
Comment (jdgriffith) [inline src/order.ts:173]: Eyeballing this seemed to show I could re-use `Order` for `replacingOrderCollection` and `childOrderStrategies`. Is that right?

File: src/order.ts [hunk 3]
```diff
@@ -1,643 +1,230 @@
...
-    accountId?: number;
-    orderActivityCollection?: {
-      activityType?: 'EXECUTION' | 'ORDER_ACTION';
-    }[];
-    replacingOrderCollection?: any[];
-    childOrderStrategies?: any[];
-    statusDescription?: string;
+    activityType?: ActivityType;
   }[];
+  replacingOrderCollection?: SavedOrder[];
```
Comment (jdgriffith) [inline src/order.ts:226]: Same idea here: `SavedOrder` could be re-used for `replacingOrderCollection` and also `childOrderStrategies`. Can you confirm this is correct?

**Comment by MorpheusNephew** [comment]
Looking at the schema provided on get saved order (https://developer.tdameritrade.com/account-access/apis/get/accounts/%7BaccountId%7D/savedorders/%7BsavedOrderId%7D-0) returns this

```
//SavedOrderGet:
{
  "session": {
    "type": "string",
    "enum": [
      "NORMAL",
      "AM",
      "PM",
      "SEAMLESS"
    ]
  },
  "duration": {
    "type": "string",
    "enum": [
      "DAY",
      "GOOD_TILL_CANCEL",
      "FILL_OR_KILL"
    ]
  },
  "orderType": {
    "type": "string",
    "enum": [
      "MARKET",
      "LIMIT",
      "STOP",
      "STOP_LIMIT",
      "TRAILING_STOP",
      "MARKET_ON_CLOSE",
      "EXERCISE",
      "TRAILING_STOP_LIMIT",
      "NET_DEBIT",
      "NET_CREDIT",
      "NET_ZERO"
    ]
  },
  "cancelTime": {
    "type": "object",
    "properties": {
      "date": {
        "type": "string"
      },
      "shortFormat": {
        "type": "boolean",
        "default": false
      }
    }
  },
  "complexOrderStrategyType": {
    "type": "string",
    "enum": [
      "NONE",
      "COVERED",
      "VERTICAL",
      "BACK_RATIO",
      "CALENDAR",
      "DIAGONAL",
      "STRADDLE",
      "STRANGLE",
      "COLLAR_SYNTHETIC",
      "BUTTERFLY",
      "CONDOR",
      "IRON_CONDOR",
      "VERTICAL_ROLL",
      "COLLAR_WITH_STOCK",
      "DOUBLE_DIAGONAL",
      "UNBALANCED_BUTTERFLY",
      "UNBALANCED_CONDOR",
      "UNBALANCED_IRON_CONDOR",
      "UNBALANCED_VERTICAL_ROLL",
      "CUSTOM"
    ]
  },
  "quantity": {
    "type": "number",
    "format": "double"
  },
  "filledQuantity": {
    "type": "number",
    "format": "double"
  },
  "remainingQuantity": {
    "type": "number",
    "format": "double"
  },
  "requestedDestination": {
    "type": "string",
    "enum": [
      "INET",
      "ECN_ARCA",
      "CBOE",
      "AMEX",
      "PHLX",
      "ISE",
      "BOX",
      "NYSE",
      "NASDAQ",
      "BATS",
      "C2",
      "AUTO"
    ]
  },
  "destinationLinkName": {
    "type": "string"
  },
  "releaseTime": {
    "type": "string",
    "format": "date-time"
  },
  "stopPrice": {
    "type": "number",
    "format": "double"
  },
  "stopPriceLinkBasis": {
    "type": "string",
    "enum": [
      "MANUAL",
      "BASE",
      "TRIGGER",
      "LAST",
      "BID",
      "ASK",
      "ASK_BID",
      "MARK",
      "AVERAGE"
    ]
  },
  "stopPriceLinkType": {
    "type": "string",
    "enum": [
      "VALUE",
      "PERCENT",
      "TICK"
    ]
  },
  "stopPriceOffset": {
    "type": "number",
    "format": "double"
  },
  "stopType": {
    "type": "string",
    "enum": [
      "STANDARD",
      "BID",
      "ASK",
      "LAST",
      "MARK"
    ]
  },
  "priceLinkBasis": {
    "type": "string",
    "enum": [
      "MANUAL",
      "BASE",
      "TRIGGER",
      "LAST",
      "BID",
      "ASK",
      "ASK_BID",
      "MARK",
      "AVERAGE"
    ]
  },
  "priceLinkType": {
    "type": "string",
    "enum": [
      "VALUE",
      "PERCENT",
      "TICK"
    ]
  },
  "price": {
    "type": "number",
    "format": "double"
  },
  "taxLotMethod": {
    "type": "string",
    "enum": [
      "FIFO",
      "LIFO",
      "HIGH_COST",
      "LOW_COST",
      "AVERAGE_COST",
      "SPECIFIC_LOT"
    ]
  },
  "orderLegCollection": {
    "type": "array",
    "xml": {
      "name": "orderLeg",
      "wrapped": true
    },
    "items": {
      "type": "object",
      "properties": {
        "orderLegType": {
          "type": "string",
          "enum": [
            "EQUITY",
            "OPTION",
            "INDEX",
            "MUTUAL_FUND",
            "CASH_EQUIVALENT",
            "FIXED_INCOME",
            "CURRENCY"
          ]
        },
        "legId": {
          "type": "integer",
          "format": "int64"
        },
        "instrument": {
          "type": "object",
          "discriminator": "assetType",
          "properties": {
            "assetType": {
              "type": "string",
              "enum": [
                "EQUITY",
                "OPTION",
                "INDEX",
                "MUTUAL_FUND",
                "CASH_EQUIVALENT",
                "FIXED_INCOME",
                "CURRENCY"
              ]
            },
            "cusip": {
              "type": "string"
            },
            "symbol": {
              "type": "string"
            },
            "description": {
              "type": "string"
            }
          }
        },
        "instruction": {
          "type": "string",
          "enum": [
            "BUY",
            "SELL",
            "BUY_TO_COVER",
            "SELL_SHORT",
            "BUY_TO_OPEN",
            "BUY_TO_CLOSE",
            "SELL_TO_OPEN",
            "SELL_TO_CLOSE",
            "EXCHANGE"
          ]
        },
        "positionEffect": {
          "type": "string",
          "enum": [
            "OPENING",
            "CLOSING",
            "AUTOMATIC"
          ]
        },
        "quantity": {
          "type": "number",
          "format": "double"
        },
        "quantityType": {
          "type": "string",
          "enum": [
            "ALL_SHARES",
            "DOLLARS",
            "SHARES"
          ]
        }
      }
    }
  },
  "activationPrice": {
    "type": "number",
    "format": "double"
  },
  "specialInstruction": {
    "type": "string",
    "enum": [
      "ALL_OR_NONE",
      "DO_NOT_REDUCE",
      "ALL_OR_NONE_DO_NOT_REDUCE"
    ]
  },
  "orderStrategyType": {
    "type": "string",
    "enum": [
      "SINGLE",
      "OCO",
      "TRIGGER"
    ]
  },
  "orderId": {
    "type": "integer",
    "format": "int64"
  },
  "cancelable": {
    "type": "boolean",
    "default": false
  },
  "editable": {
    "type": "boolean",
    "default": false
  },
  "status": {
    "type": "string",
    "enum": [
      "AWAITING_PARENT_ORDER",
      "AWAITING_CONDITION",
      "AWAITING_MANUAL_REVIEW",
      "ACCEPTED",
      "AWAITING_UR_OUT",
      "PENDING_ACTIVATION",
      "QUEUED",
      "WORKING",
      "REJECTED",
      "PENDING_CANCEL",
      "CANCELED",
      "PENDING_REPLACE",
      "REPLACED",
      "FILLED",
      "EXPIRED"
    ]
  },
  "enteredTime": {
    "type": "string",
    "format": "date-time"
  },
  "closeTime": {
    "type": "string",
    "format": "date-time"
  },
  "tag": {
    "type": "string"
  },
  "accountId": {
    "type": "integer",
    "format": "int64"
  },
  "orderActivityCollection": {
    "type": "array",
    "xml": {
      "name": "orderActivity",
      "wrapped": true
    },
    "items": {
      "type": "object",
      "discriminator": "activityType",
      "properties": {
        "activityType": {
          "type": "string",
          "enum": [
            "EXECUTION",
            "ORDER_ACTION"
          ]
        }
      }
    }
  },
  "replacingOrderCollection": {
    "type": "array",
    "xml": {
      "name": "replacingOrder",
      "wrapped": true
    },
    "items": {
      "type": "object",
      "properties": {
        "session": {
          "type": "string",
          "enum": [
            "NORMAL",
            "AM",
            "PM",
            "SEAMLESS"
          ]
        },
        "duration": {
          "type": "string",
          "enum": [
            "DAY",
            "GOOD_TILL_CANCEL",
            "FILL_OR_KILL"
          ]
        },
        "orderType": {
          "type": "string",
          "enum": [
            "MARKET",
            "LIMIT",
            "STOP",
            "STOP_LIMIT",
            "TRAILING_STOP",
            "MARKET_ON_CLOSE",
            "EXERCISE",
            "TRAILING_STOP_LIMIT",
            "NET_DEBIT",
            "NET_CREDIT",
            "NET_ZERO"
          ]
        },
        "cancelTime": {
          "type": "object",
          "properties": {
            "date": {
              "type": "string"
            },
            "shortFormat": {
              "type": "boolean",
              "default": false
            }
          }
        },
        "complexOrderStrategyType": {
          "type": "string",
          "enum": [
            "NONE",
            "COVERED",
            "VERTICAL",
            "BACK_RATIO",
            "CALENDAR",
            "DIAGONAL",
            "STRADDLE",
            "STRANGLE",
            "COLLAR_SYNTHETIC",
            "BUTTERFLY",
            "CONDOR",
            "IRON_CONDOR",
            "VERTICAL_ROLL",
            "COLLAR_WITH_STOCK",
            "DOUBLE_DIAGONAL",
            "UNBALANCED_BUTTERFLY",
            "UNBALANCED_CONDOR",
            "UNBALANCED_IRON_CONDOR",
            "UNBALANCED_VERTICAL_ROLL",
            "CUSTOM"
          ]
        },
        "quantity": {
          "type": "number",
          "format": "double"
        },
        "filledQuantity": {
          "type": "number",
          "format": "double"
        },
        "remainingQuantity": {
          "type": "number",
          "format": "double"
        },
        "requestedDestination": {
          "type": "string",
          "enum": [
            "INET",
            "ECN_ARCA",
            "CBOE",
            "AMEX",
            "PHLX",
            "ISE",
            "BOX",
            "NYSE",
            "NASDAQ",
            "BATS",
            "C2",
            "AUTO"
          ]
        },
        "destinationLinkName": {
          "type": "string"
        },
        "releaseTime": {
          "type": "string",
          "format": "date-time"
        },
        "stopPrice": {
          "type": "number",
          "format": "double"
        },
        "stopPriceLinkBasis": {
          "type": "string",
          "enum": [
            "MANUAL",
            "BASE",
            "TRIGGER",
            "LAST",
            "BID",
            "ASK",
            "ASK_BID",
            "MARK",
            "AVERAGE"
          ]
        },
        "stopPriceLinkType": {
          "type": "string",
          "enum": [
            "VALUE",
            "PERCENT",
            "TICK"
          ]
        },
        "stopPriceOffset": {
          "type": "number",
          "format": "double"
        },
        "stopType": {
          "type": "string",
          "enum": [
            "STANDARD",
            "BID",
            "ASK",
            "LAST",
            "MARK"
          ]
        },
        "priceLinkBasis": {
          "type": "string",
          "enum": [
            "MANUAL",
            "BASE",
            "TRIGGER",
            "LAST",
            "BID",
            "ASK",
            "ASK_BID",
            "MARK",
            "AVERAGE"
          ]
        },
        "priceLinkType": {
          "type": "string",
          "enum": [
            "VALUE",
            "PERCENT",
            "TICK"
          ]
        },
        "price": {
          "type": "number",
          "format": "double"
        },
        "taxLotMethod": {
          "type": "string",
          "enum": [
            "FIFO",
            "LIFO",
            "HIGH_COST",
            "LOW_COST",
            "AVERAGE_COST",
            "SPECIFIC_LOT"
          ]
        },
        "orderLegCollection": {
          "type": "array",
          "xml": {
            "name": "orderLeg",
            "wrapped": true
          },
          "items": {
            "type": "object",
            "properties": {
              "orderLegType": {
                "type": "string",
                "enum": [
                  "EQUITY",
                  "OPTION",
                  "INDEX",
                  "MUTUAL_FUND",
                  "CASH_EQUIVALENT",
                  "FIXED_INCOME",
                  "CURRENCY"
                ]
              },
              "legId": {
                "type": "integer",
                "format": "int64"
              },
              "instrument": {
                "type": "object",
                "discriminator": "assetType",
                "properties": {
                  "assetType": {
                    "type": "string",
                    "enum": [
                      "EQUITY",
                      "OPTION",
                      "INDEX",
                      "MUTUAL_FUND",
                      "CASH_EQUIVALENT",
                      "FIXED_INCOME",
                      "CURRENCY"
                    ]
                  },
                  "cusip": {
                    "type": "string"
                  },
                  "symbol": {
                    "type": "string"
                  },
                  "description": {
                    "type": "string"
                  }
                }
              },
              "instruction": {
                "type": "string",
                "enum": [
                  "BUY",
                  "SELL",
                  "BUY_TO_COVER",
                  "SELL_SHORT",
                  "BUY_TO_OPEN",
                  "BUY_TO_CLOSE",
                  "SELL_TO_OPEN",
                  "SELL_TO_CLOSE",
                  "EXCHANGE"
                ]
              },
              "positionEffect": {
                "type": "string",
                "enum": [
                  "OPENING",
                  "CLOSING",
                  "AUTOMATIC"
                ]
              },
              "quantity": {
                "type": "number",
                "format": "double"
              },
              "quantityType": {
                "type": "string",
                "enum": [
                  "ALL_SHARES",
                  "DOLLARS",
                  "SHARES"
                ]
              }
            }
          }
        },
        "activationPrice": {
          "type": "number",
          "format": "double"
        },
        "specialInstruction": {
          "type": "string",
          "enum": [
            "ALL_OR_NONE",
            "DO_NOT_REDUCE",
            "ALL_OR_NONE_DO_NOT_REDUCE"
          ]
        },
        "orderStrategyType": {
          "type": "string",
          "enum": [
            "SINGLE",
            "OCO",
            "TRIGGER"
          ]
        },
        "orderId": {
          "type": "integer",
          "format": "int64"
        },
        "cancelable": {
          "type": "boolean",
          "default": false
        },
        "editable": {
          "type": "boolean",
          "default": false
        },
        "status": {
          "type": "string",
          "enum": [
            "AWAITING_PARENT_ORDER",
            "AWAITING_CONDITION",
            "AWAITING_MANUAL_REVIEW",
            "ACCEPTED",
            "AWAITING_UR_OUT",
            "PENDING_ACTIVATION",
            "QUEUED",
            "WORKING",
            "REJECTED",
            "PENDING_CANCEL",
            "CANCELED",
            "PENDING_REPLACE",
            "REPLACED",
            "FILLED",
            "EXPIRED"
          ]
        },
        "enteredTime": {
          "type": "string",
          "format": "date-time"
        },
        "closeTime": {
          "type": "string",
          "format": "date-time"
        },
        "tag": {
          "type": "string"
        },
        "accountId": {
          "type": "integer",
          "format": "int64"
        },
        "orderActivityCollection": {
          "type": "array",
          "xml": {
            "name": "orderActivity",
            "wrapped": true
          },
          "items": {
            "type": "object",
            "discriminator": "activityType",
            "properties": {
              "activityType": {
                "type": "string",
                "enum": [
                  "EXECUTION",
                  "ORDER_ACTION"
                ]
              }
            }
          }
        },
        "replacingOrderCollection": {
          "type": "array",
          "xml": {
            "name": "replacingOrder",
            "wrapped": true
          }
        },
        "childOrderStrategies": {
          "type": "array",
          "xml": {
            "name": "childOrder",
            "wrapped": true
          }
        },
        "statusDescription": {
          "type": "string"
        }
      }
    }
  },
  "childOrderStrategies": {
    "type": "array",
    "xml": {
      "name": "childOrder",
      "wrapped": true
    },
    "items": {
      "type": "object",
      "properties": {
        "session": {
          "type": "string",
          "enum": [
            "NORMAL",
            "AM",
            "PM",
            "SEAMLESS"
          ]
        },
        "duration": {
          "type": "string",
          "enum": [
            "DAY",
            "GOOD_TILL_CANCEL",
            "FILL_OR_KILL"
          ]
        },
        "orderType": {
          "type": "string",
          "enum": [
            "MARKET",
            "LIMIT",
            "STOP",
            "STOP_LIMIT",
            "TRAILING_STOP",
            "MARKET_ON_CLOSE",
            "EXERCISE",
            "TRAILING_STOP_LIMIT",
            "NET_DEBIT",
            "NET_CREDIT",
            "NET_ZERO"
          ]
        },
        "cancelTime": {
          "type": "object",
          "properties": {
            "date": {
              "type": "string"
            },
            "shortFormat": {
              "type": "boolean",
              "default": false
            }
          }
        },
        "complexOrderStrategyType": {
          "type": "string",
          "enum": [
            "NONE",
            "COVERED",
            "VERTICAL",
            "BACK_RATIO",
            "CALENDAR",
            "DIAGONAL",
            "STRADDLE",
            "STRANGLE",
            "COLLAR_SYNTHETIC",
            "BUTTERFLY",
            "CONDOR",
            "IRON_CONDOR",
            "VERTICAL_ROLL",
            "COLLAR_WITH_STOCK",
            "DOUBLE_DIAGONAL",
            "UNBALANCED_BUTTERFLY",
            "UNBALANCED_CONDOR",
            "UNBALANCED_IRON_CONDOR",
            "UNBALANCED_VERTICAL_ROLL",
            "CUSTOM"
          ]
        },
        "quantity": {
          "type": "number",
          "format": "double"
        },
        "filledQuantity": {
          "type": "number",
          "format": "double"
        },
        "remainingQuantity": {
          "type": "number",
          "format": "double"
        },
        "requestedDestination": {
          "type": "string",
          "enum": [
            "INET",
            "ECN_ARCA",
            "CBOE",
            "AMEX",
            "PHLX",
            "ISE",
            "BOX",
            "NYSE",
            "NASDAQ",
            "BATS",
            "C2",
            "AUTO"
          ]
        },
        "destinationLinkName": {
          "type": "string"
        },
        "releaseTime": {
          "type": "string",
          "format": "date-time"
        },
        "stopPrice": {
          "type": "number",
          "format": "double"
        },
        "stopPriceLinkBasis": {
          "type": "string",
          "enum": [
            "MANUAL",
            "BASE",
            "TRIGGER",
            "LAST",
            "BID",
            "ASK",
            "ASK_BID",
            "MARK",
            "AVERAGE"
          ]
        },
        "stopPriceLinkType": {
          "type": "string",
          "enum": [
            "VALUE",
            "PERCENT",
            "TICK"
          ]
        },
        "stopPriceOffset": {
          "type": "number",
          "format": "double"
        },
        "stopType": {
          "type": "string",
          "enum": [
            "STANDARD",
            "BID",
            "ASK",
            "LAST",
            "MARK"
          ]
        },
        "priceLinkBasis": {
          "type": "string",
          "enum": [
            "MANUAL",
            "BASE",
            "TRIGGER",
            "LAST",
            "BID",
            "ASK",
            "ASK_BID",
            "MARK",
            "AVERAGE"
          ]
        },
        "priceLinkType": {
          "type": "string",
          "enum": [
            "VALUE",
            "PERCENT",
            "TICK"
          ]
        },
        "price": {
          "type": "number",
          "format": "double"
        },
        "taxLotMethod": {
          "type": "string",
          "enum": [
            "FIFO",
            "LIFO",
            "HIGH_COST",
            "LOW_COST",
            "AVERAGE_COST",
            "SPECIFIC_LOT"
          ]
        },
        "orderLegCollection": {
          "type": "array",
          "xml": {
            "name": "orderLeg",
            "wrapped": true
          },
          "items": {
            "type": "object",
            "properties": {
              "orderLegType": {
                "type": "string",
                "enum": [
                  "EQUITY",
                  "OPTION",
                  "INDEX",
                  "MUTUAL_FUND",
                  "CASH_EQUIVALENT",
                  "FIXED_INCOME",
                  "CURRENCY"
                ]
              },
              "legId": {
                "type": "integer",
                "format": "int64"
              },
              "instrument": {
                "type": "object",
                "discriminator": "assetType",
                "properties": {
                  "assetType": {
                    "type": "string",
                    "enum": [
                      "EQUITY",
                      "OPTION",
                      "INDEX",
                      "MUTUAL_FUND",
                      "CASH_EQUIVALENT",
                      "FIXED_INCOME",
                      "CURRENCY"
                    ]
                  },
                  "cusip": {
                    "type": "string"
                  },
                  "symbol": {
                    "type": "string"
                  },
                  "description": {
                    "type": "string"
                  }
                }
              },
              "instruction": {
                "type": "string",
                "enum": [
                  "BUY",
                  "SELL",
                  "BUY_TO_COVER",
                  "SELL_SHORT",
                  "BUY_TO_OPEN",
                  "BUY_TO_CLOSE",
                  "SELL_TO_OPEN",
                  "SELL_TO_CLOSE",
                  "EXCHANGE"
                ]
              },
              "positionEffect": {
                "type": "string",
                "enum": [
                  "OPENING",
                  "CLOSING",
                  "AUTOMATIC"
                ]
              },
              "quantity": {
                "type": "number",
                "format": "double"
              },
              "quantityType": {
                "type": "string",
                "enum": [
                  "ALL_SHARES",
                  "DOLLARS",
                  "SHARES"
                ]
              }
            }
          }
        },
        "activationPrice": {
          "type": "number",
          "format": "double"
        },
        "specialInstruction": {
          "type": "string",
          "enum": [
            "ALL_OR_NONE",
            "DO_NOT_REDUCE",
            "ALL_OR_NONE_DO_NOT_REDUCE"
          ]
        },
        "orderStrategyType": {
          "type": "string",
          "enum": [
            "SINGLE",
            "OCO",
            "TRIGGER"
          ]
        },
        "orderId": {
          "type": "integer",
          "format": "int64"
        },
        "cancelable": {
          "type": "boolean",
          "default": false
        },
        "editable": {
          "type": "boolean",
          "default": false
        },
        "status": {
          "type": "string",
          "enum": [
            "AWAITING_PARENT_ORDER",
            "AWAITING_CONDITION",
            "AWAITING_MANUAL_REVIEW",
            "ACCEPTED",
            "AWAITING_UR_OUT",
            "PENDING_ACTIVATION",
            "QUEUED",
            "WORKING",
            "REJECTED",
            "PENDING_CANCEL",
            "CANCELED",
            "PENDING_REPLACE",
            "REPLACED",
            "FILLED",
            "EXPIRED"
          ]
        },
        "enteredTime": {
          "type": "string",
          "format": "date-time"
        },
        "closeTime": {
          "type": "string",
          "format": "date-time"
        },
        "tag": {
          "type": "string"
        },
        "accountId": {
          "type": "integer",
          "format": "int64"
        },
        "orderActivityCollection": {
          "type": "array",
          "xml": {
            "name": "orderActivity",
            "wrapped": true
          },
          "items": {
            "type": "object",
            "discriminator": "activityType",
            "properties": {
              "activityType": {
                "type": "string",
                "enum": [
                  "EXECUTION",
                  "ORDER_ACTION"
                ]
              }
            }
          }
        },
        "replacingOrderCollection": {
          "type": "array",
          "xml": {
            "name": "replacingOrder",
            "wrapped": true
          }
        },
        "childOrderStrategies": {
          "type": "array",
          "xml": {
            "name": "childOrder",
            "wrapped": true
          }
        },
        "statusDescription": {
          "type": "string"
        }
      }
    }
  },
  "statusDescription": {
    "type": "string"
  },
  "savedOrderId": {
    "type": "integer",
    "format": "int64"
  },
  "savedTime": {
    "type": "string",
    "format": "date-time"
  }
}

//The class <Instrument> has the following subclasses: 
//-Equity
//-FixedIncome
//-MutualFund
//-CashEquivalent
//-Option
//schemas for each are listed below: 

//Equity:
{
  "assetType": {
    "type": "string",
    "enum": [
      "EQUITY",
      "OPTION",
      "INDEX",
      "MUTUAL_FUND",
      "CASH_EQUIVALENT",
      "FIXED_INCOME",
      "CURRENCY"
    ]
  },
  "cusip": {
    "type": "string"
  },
  "symbol": {
    "type": "string"
  },
  "description": {
    "type": "string"
  }
}

//OR

//FixedIncome:
{
  "assetType": {
    "type": "string",
    "enum": [
      "EQUITY",
      "OPTION",
      "INDEX",
      "MUTUAL_FUND",
      "CASH_EQUIVALENT",
      "FIXED_INCOME",
      "CURRENCY"
    ]
  },
  "cusip": {
    "type": "string"
  },
  "symbol": {
    "type": "string"
  },
  "description": {
    "type": "string"
  },
  "maturityDate": {
    "type": "string",
    "format": "date-time"
  },
  "variableRate": {
    "type": "number",
    "format": "double"
  },
  "factor": {
    "type": "number",
    "format": "double"
  }
}

//OR

//MutualFund:
{
  "assetType": {
    "type": "string",
    "enum": [
      "EQUITY",
      "OPTION",
      "INDEX",
      "MUTUAL_FUND",
      "CASH_EQUIVALENT",
      "FIXED_INCOME",
      "CURRENCY"
    ]
  },
  "cusip": {
    "type": "string"
  },
  "symbol": {
    "type": "string"
  },
  "description": {
    "type": "string"
  },
  "type": {
    "type": "string",
    "enum": [
      "NOT_APPLICABLE",
      "OPEN_END_NON_TAXABLE",
      "OPEN_END_TAXABLE",
      "NO_LOAD_NON_TAXABLE",
      "NO_LOAD_TAXABLE"
    ]
  }
}

//OR

//CashEquivalent:
{
  "assetType": {
    "type": "string",
    "enum": [
      "EQUITY",
      "OPTION",
      "INDEX",
      "MUTUAL_FUND",
      "CASH_EQUIVALENT",
      "FIXED_INCOME",
      "CURRENCY"
    ]
  },
  "cusip": {
    "type": "string"
  },
  "symbol": {
    "type": "string"
  },
  "description": {
    "type": "string"
  },
  "type": {
    "type": "string",
    "enum": [
      "SAVINGS",
      "MONEY_MARKET_FUND"
    ]
  }
}

//OR

//Option:
{
  "assetType": {
    "type": "string",
    "enum": [
      "EQUITY",
      "OPTION",
      "INDEX",
      "MUTUAL_FUND",
      "CASH_EQUIVALENT",
      "FIXED_INCOME",
      "CURRENCY"
    ]
  },
  "cusip": {
    "type": "string"
  },
  "symbol": {
    "type": "string"
  },
  "description": {
    "type": "string"
  },
  "type": {
    "type": "string",
    "enum": [
      "VANILLA",
      "BINARY",
      "BARRIER"
    ]
  },
  "putCall": {
    "type": "string",
    "enum": [
      "PUT",
      "CALL"
    ]
  },
  "underlyingSymbol": {
    "type": "string"
  },
  "optionMultiplier": {
    "type": "integer",
    "format": "int32"
  },
  "optionDeliverables": {
    "type": "array",
    "xml": {
      "name": "optionDeliverable",
      "wrapped": true
    },
    "items": {
      "type": "object",
      "properties": {
        "symbol": {
          "type": "string"
        },
        "deliverableUnits": {
          "type": "number",
          "format": "double"
        },
        "currencyType": {
          "type": "string",
          "enum": [
            "USD",
            "CAD",
            "EUR",
            "JPY"
          ]
        },
        "assetType": {
          "type": "string",
          "enum": [
            "EQUITY",
            "OPTION",
            "INDEX",
            "MUTUAL_FUND",
            "CASH_EQUIVALENT",
            "FIXED_INCOME",
            "CURRENCY"
          ]
        }
      }
    }
  }
}

//The class <OrderActivity> has the following subclasses: 
//-Execution
//schemas for each are listed below: 

//Execution:
{
  "activityType": {
    "type": "string",
    "enum": [
      "EXECUTION",
      "ORDER_ACTION"
    ]
  },
  "executionType": {
    "type": "string",
    "enum": [
      "FILL"
    ]
  },
  "quantity": {
    "type": "number",
    "format": "double"
  },
  "orderRemainingQuantity": {
    "type": "number",
    "format": "double"
  },
  "executionLegs": {
    "type": "array",
    "xml": {
      "name": "executionLeg",
      "wrapped": true
    },
    "items": {
      "type": "object",
      "properties": {
        "legId": {
          "type": "integer",
          "format": "int32"
        },
        "quantity": {
          "type": "number",
          "format": "double"
        },
        "mismarkedQuantity": {
          "type": "number",
          "format": "double"
        },
        "price": {
          "type": "number",
          "format": "double"
        },
        "time": {
          "type": "string",
          "format": "date-time"
        }
      }
    }
  }
}
```

**Comment by MorpheusNephew** [comment]
I'll take a look and see if it's meant to represent orders and if not I'll change it back

**Comment by MorpheusNephew** [comment]
Thanks for doing this @jdgriffith, looks like your inclination was correct

**Pull request merged**